source/frontend-packages.rst

```diff
@@ -0,0 +1,198 @@
+
+* `package.json <_static/files/package.json>`__
+
+and add them to the top level of your project. You'll need to modify
+both with the correct project name.
+
+Finally, you can add these extra lines to your ``.gitignore``::
+
+    node_modules
+    */static/js/bundle.js
+    */static/css
+
+**TODO** Finish the upgrade section. It is incomplete.
+
+Using Frontend Packages
+=======================
+
+Our frontend packaging includes configuration to support the newest version of
+the Javascript language ES2015, which includes many updates including proper
+module support and an import statement. To use packages installed in your project
+you can import the default export from the project under a name of your choosing::
+
+    import $ from 'jquery';
+
+For some packages, you'll likely want to import more than one thing from it::
+
+    import { AutoBind, jQueryClass } from './helpers/jquery';
+
+Organizing Project Frontend Code
+--------------------------------
+
+Packages aren't just for the third party code you depend on. You'll
+benefit from organizing your project's own Javascript the same way. This setup begins
+with a top-level module in your project at ``static/js/index.js``. This module
+is only expected to be an initial point to import all your dependencies and do
+very basic initialization.
+
+A note for projects that use jQuery: because of the global nature of jQuery and
+the hoisting nature of ES2015 import statements, it can be very difficult to import
+jQuery in a way that makes it easily available to other scripts, such as plugins.
+We have found a double import method as follows works around these issues:
+
+.. code-block:: javascript
+
+    import $ from 'jquery'
+    /* ... additional imports ... */
+
+    window.jQuery = window.$ = require('jquery')
+
+The first, native import will ensure jQuery is loaded early as a dependency. The second
+version uses the non-native ``require()`` function, which is the NodeJS format for
+module import before ES2015. This version is just a regular line of code and so it will
+*not* be hoisted and can properly inject the window globals ``jQuery`` and ``$``.
+
+Creating And Using Modules
+--------------------------
+
+New functionality may require you organize it into its own module. Use your best
+judgement here, but consider the same sort of reasoning we use to draw lines between
+functionality in Python. You can also group related modules into a package, by placing
+them in a directory.
+
+You can import your module to load it from the ``index.js``
+
+.. code-block:: javascript
+
+    import './utils.js'
+
+The ``./`` is necessary to distinguish from modules that would be loaded from the
+standard library or NPM installed packages. These imports are simply relative paths to
+the location of the module.
+
+You only need to include modules you might consider "top-level" in ``index.js``. For
+example, if you defined a helper function ``get_csrf_token()`` in that ``utils.js``
```

Comment:
I find it a little confusing to see the function in its invocation form here. Is that the Python way to do it, or something?